src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,361 @@
+  import DataGridActions from '../ui/DataGrid/DataGridActions.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SectionTitle from '../ui/SectionTitle.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let user: User | null;
+
+  type CellRendererParams = {
+    viewDerivationGroup: (derivationGroup: DerivationGroupMetadata) => void;
+  };
+  type DerivationGroupCellRendererParams = ICellRendererParams<DerivationGroupMetadata> & CellRendererParams;
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+  }>();
+
+  let externalEventLayers: ExternalEventLayer[] | undefined;
+  let dataGrid: DataGrid<DerivationGroup>;
+  let baseColumnDefs: DataGridColumnDef<DerivationGroup>[] = [];
+
+  selectedPlanDerivationGroupIds.subscribe(() => {
+    if (baseColumnDefs.length > 0 && dataGrid) {
+      // no current way to change just a specific cell unless we add something about plan associations to the DG object,
+      //    which we don't seek to do.
+      // this does mean every update to any entry in selectedPlanDerivationGroupIds refreshes the whole column, and flashes it,
+      //    though that isn't particularly a problem and does do good to signal association complete. Also a small delay, which
+      //    buffers button smashing and repeated updates pretty well!
+      dataGrid.refreshCells();
+    }
+  });
+
+  $: externalEventLayers = $view?.definition.plan.timelines
+    .flatMap(timeline => timeline.rows)
+    .flatMap(row => row.layers)
+    .filter(layer => isExternalEventLayer(layer));
+
+  baseColumnDefs = [
+    {
+      field: 'name',
+      filter: 'string',
+      headerName: 'Derivation Group',
+      resizable: true,
+      sortable: true,
+      suppressAutoSize: false,
+      suppressSizeToFit: false,
+    },
+    {
+      field: 'source_type_id',
+      filter: 'string',
+      headerName: 'Source type',
+      resizable: true,
+      sortable: true,
+      suppressAutoSize: false,
+      suppressSizeToFit: false,
+      valueFormatter: params => {
+        const sourceTypeName = getEventSourceTypeName(params?.value, $externalSourceTypes);
+        return sourceTypeName ? sourceTypeName : '';
+      },
+    },
+    {
+      field: 'derivedEventTotal',
+      filter: 'number',
+      headerName: 'Derived Events in Derivation Group',
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      valueFormatter: params => {
+        return params?.value.length;
+      },
+    },
+    {
+      cellRenderer: (params: DerivationGroupCellRendererParams) => {
+        var input = document.createElement('input');
+        input.type = 'checkbox';
+        input.checked = $selectedPlanDerivationGroupIds.includes(params?.data?.id ?? -1);
+        input.addEventListener('click', event => {
```

Comment:
Yeah! I think your former example looks good! I think it'll be ok to keep the remaining update code.

Reply:
I tried making this fix and I was running into some type casting issues, as well as the issue that `cellEditingStopped` wouldn't fire. I looked into the docs, and [it seemed like `editable: true` needed to be set](https://www.ag-grid.com/javascript-data-grid/cell-editing/), but when that is the case, instead of an event firing when the checkbox is selected or deselected, the event is fired when a checkbox is double clicked, at which point it turns into an editable checkbox that doesn't actually do anything. As such, we are leaving it as is, as [there is precedent](https://www.ag-grid.com/javascript-data-grid/component-cell-renderer/) even in the documentation for defining an HTML element and returning it in the `cellRenderer` as we have done, but we have moved the event listener to a separate function for the sake of decluttering. It seems like the primary issue is that this was a very convoluted function, so this simplification should address that issue.

Reply:
See [this commit](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/5756e9fe9c0f49dd6f0454df66f03882053a7f0a).

Reply:
Fully fixed [now](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/789cbfc1e3e1564544be90d5ef383e3be930e69a).